### PR TITLE
Adding lesson 04 files

### PR DESCRIPTION
**Pull request by jineehan**
Adding lesson 04 files.

### CODE REVIEW

File: students/jinee_han/lesson04/Mail2.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        inputValue = input("Enter a full name. (Type 'list' to see the donor list)")
+
+    if not donor_present(inputValue):
+        add_donor(inputValue)
+
+    donation_name = inputValue
+    donation_amount = float(input("Please enter the donation amount: "))
+    add_donation_amount(donation_name, donation_amount)
+    format_thank_you_note(donation_name, donation_amount)
+
```
Comment (charlesrobison) [inline students/jinee_han/lesson04/Mail2.py]: Per PEP8, you'll want 2 blank lines between functions - here and below

File: students/jinee_han/lesson04/Mail2.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    f.close()
+
+def format_thank_you_note(donator, donation_amount):
+    '''
+    Format the thank you note before writing to file
+    :param donator: The current donator
+    :param donation_amount: The donation amount
+    :return: nothing
+    '''
+    lstObj = []
```
Comment (charlesrobison) [inline students/jinee_han/lesson04/Mail2.py]: This should be snake_case: lst_obj, or list_object even better

File: students/jinee_han/lesson04/Mail2.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+            "Justin Bieber": [1663.23, 4300.87, 10432.0],
+            "Will Smith": [43.23, 4000.07, 183423.2]
+            }
+
+# Display options
+prompt = "\n".join(("Welcome to the donor list!",
+          "Please choose from below options:",
+          "1 - Send a Thank you",
+          "2 - Create a Report",
+          "3 - Exit",
```
Comment (charlesrobison) [inline students/jinee_han/lesson04/Mail2.py]: Missing option to create letters for all donors, saving to disk

File: students/jinee_han/lesson04/Mail2.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+input_dict = {1: send_thank_you_note, 2: create_report, 3: exit_program}
+
+def main():
+    while True:
+        response = int(input(prompt))  # continuously collect user selection
+        input_dict.get(response)()
+
+if __name__ == "__main__":
+    main()
```
Comment (charlesrobison) [inline students/jinee_han/lesson04/Mail2.py]: This is breaking when adding a new donor

File: students/jinee_han/lesson04/Trigrams.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    '''
+    file = open(filename, "r")
+    word_collection = []
+    for line in file:
+        line = line.strip('\n')
+        line = re.sub(r'[^\w\s]',' ',line)
+        for word in line.split():
+            word_collection.append(word.lower())
+    return word_collection
+
```
Comment (charlesrobison) [inline students/jinee_han/lesson04/Trigrams.py]: Same as with mailroom file - 2 blank lines between functions.

File: students/jinee_han/lesson04/Trigrams.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        filename = sys.argv[1]
+    except IndexError:
+        print("You must pass in a filename!")
+        sys.exit(1)
+
+    word_collection = read_and_clean_text(filename)
+    dict_text = build_trigram_dict(word_collection)
+    new_text = create_text(dict_text, max_length=250)
+    print(new_text.capitalize())
+
```
Comment (charlesrobison) [inline students/jinee_han/lesson04/Trigrams.py]: Works well!  Nice job

**Comment by jineehan** [comment]
Hello Charles,

Thank you for the previous PR review. I created a new PR to address the
missing functionality/spacing issues. Here is a link:
https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/331

Please let me know if I should submit this PR link elsewhere.

Thanks,
Jinee

**Pull request merged**